rele/worker.py

```diff
@@ -88,11 +88,19 @@ def stop(self, signal=None, frame=None):
         :param signal: Needed for `signal.signal <https://docs.python.org/3/library/signal.html#signal.signal>`_  # noqa
         :param frame: Needed for `signal.signal <https://docs.python.org/3/library/signal.html#signal.signal>`_  # noqa
         """
+
+        logger.info(f"[rele] {signal} received, stopping workers")
+
         run_middleware_hook("pre_worker_stop", self._subscriptions)
         for future in self._futures.values():
             future.cancel()
 
+        logger.info("[rele] future cancelled")
+
         run_middleware_hook("post_worker_stop")
+
+        logger.info("[rele] worker stopped")
```

Comment:
Are we sure that self._futures is always populated? It's going to log either it has cancelled a future or not.
Also I think that we are poluting the method with intermediate logs.

I have checked the `run_middleware_hook` function and it doesn't have any log.

What do you think about logging inside `run_middleware_hook` at the beginning and the end, including the `hook_name`?